src/ragged/_spec_set_functions.py

```diff
@@ -53,20 +53,28 @@ def unique_all(x: array, /) -> tuple[array, array, array, array]:
     """
 
     if isinstance(x, ragged.array):
-        if len(x) == 1:
+        if x.ndim == 0:
             return unique_all_result(
-                values=ragged.array(x),
+                values=ragged.array(np.unique(x._impl, equal_nan=False)),  # pylint: disable=W0212
                 indices=ragged.array([0]),
                 inverse_indices=ragged.array([0]),
                 counts=ragged.array([1]),
             )
         else:
-            x_flat = ak.ravel(x._impl)
+            x_flat = ak.ravel(x._impl)  # pylint: disable=W0212
+            if isinstance(x_flat.layout, ak.contents.EmptyArray):  # pylint: disable=E1101
+                return unique_all_result(
+                    values=ragged.array([]),
+                    indices=ragged.array([]),
+                    inverse_indices=ragged.array([]),
+                    counts=ragged.array([]),
```

Comment:
You need to control the dtypes of these arrays:

```suggestion
                    values=ragged.array(np.empty(0, x.dtype)),
                    indices=ragged.array(np.empty(0, np.int64)),
                    inverse_indices=ragged.array(np.empty(0, np.int64)),
                    counts=ragged.array(np.empty(0, np.int64)),
```

Something to check: are the dtypes the same when they're not empty? That is, I'm not confusing int32 and nit64, am I?